llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -935,14 +964,13 @@ static VSETVLIInfo computeInfoForInstr(const MachineInstr &MI, uint64_t TSFlags,
   // AVL operand with the AVL of the defining vsetvli.  We avoid general
   // register AVLs to avoid extending live ranges without being sure we can
   // kill the original source reg entirely.
-  if (InstrInfo.hasAVLReg() && InstrInfo.getAVLReg().isVirtual()) {
-    MachineInstr *DefMI = MRI->getVRegDef(InstrInfo.getAVLReg());
-    if (DefMI && isVectorConfigInstr(*DefMI)) {
+  if (InstrInfo.hasAVLReg()) {
+    MachineInstr *DefMI = MRI->getUniqueVRegDef(InstrInfo.getAVLReg());
+    if (isVectorConfigInstr(*DefMI)) {
```

Comment:
getUniqueVRegDef may return a nullptr if none is found, although a valid program should always have a VRegDef here.

Reply:
Woops yes, that should have an assert that DefMI isn't null underneath there. And then we can remove it after we move the definitions into VSETVLIInfo.